test/gcs/main_test.go

```diff
@@ -0,0 +1,171 @@
+	"github.com/sirupsen/logrus"
+
+	"github.com/Microsoft/hcsshim/internal/guest/runtime"
+	"github.com/Microsoft/hcsshim/internal/guest/runtime/hcsv2"
+	"github.com/Microsoft/hcsshim/internal/guest/runtime/runc"
+	"github.com/Microsoft/hcsshim/internal/guest/transport"
+	"github.com/Microsoft/hcsshim/internal/guestpath"
+	"github.com/Microsoft/hcsshim/pkg/securitypolicy"
+
+	testflag "github.com/Microsoft/hcsshim/test/internal/flag"
+	"github.com/Microsoft/hcsshim/test/internal/require"
+)
+
+const (
+	featureCRI        = "CRI"
+	featureStandalone = "StandAlone"
+)
+
+var _securityPolicy string
+
+var allFeatures = []string{
+	featureCRI,
+	featureStandalone,
+}
+
+// flags
+var (
+	flagFeatures      = testflag.NewFeatureFlag(allFeatures)
+	flagJoinGCSCgroup = flag.Bool(
+		"join-gcs-cgroup",
+		false,
+		"If true, join the same cgroup as the gcs daemon, `/gcs`",
+	)
+	flagRootfsPath = flag.String(
+		"rootfs-path",
+		"/run/rootfs",
+		"The path on the uVM of the unpacked rootfs to use for the containers",
+	)
+	flagSandboxPause = flag.Bool(
+		"pause-sandbox",
+		false,
+		"Use `/pause` as the sandbox container command",
+	)
+)
+
+func init() {
+	var err error
+	p := securitypolicy.NewOpenDoorPolicy()
+	_securityPolicy, err = p.EncodeToString()
+	if err != nil {
+		// really should not get here ...
+		log.Fatal("could not encode open door policy to string: %w", err)
+	}
+}
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+
+	if err := setup(); err != nil {
+		logrus.WithError(err).Fatal("could not set up testing")
+	}
+
+	os.Exit(m.Run())
+}
+
+func setup() (err error) {
+	os.MkdirAll(guestpath.LCOWRootPrefixInUVM, 0755)
+
+	if vf := flag.Lookup("test.v"); vf != nil {
+		if vf.Value.String() == strconv.FormatBool(true) {
+			logrus.SetLevel(logrus.DebugLevel)
+		} else {
+			logrus.SetLevel(logrus.ErrorLevel)
+		}
+	}
+
+	// should already start gcs cgroup
+	if !*flagJoinGCSCgroup {
+		gcsControl, err := cgroups.Load(cgroups.V1, cgroups.StaticPath("/"))
+		if err != nil {
+			return fmt.Errorf("failed to load root cgroup: %w", err)
+		}
+		if err := gcsControl.Add(cgroups.Process{Pid: os.Getpid()}); err != nil {
+			return fmt.Errorf("failed join root cgroup: %w", err)
+		}
+		logrus.Debug("joined root cgroup")
+	}
+
+	// initialize runtime
+	rt, err := _getRuntime()
+	if err != nil {
+		return err
+	}
+
+	// check policy will be parsed properly
+	if _, err = _getHost(rt, getTransport()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+//
+// host and runtime management
+//
+
+func getTestState(ctx context.Context, t testing.TB) (*hcsv2.Host, runtime.Runtime) {
+	rt := getRuntime(ctx, t)
+
+	return getHost(ctx, t, rt), rt
+}
+
+func getHost(_ context.Context, t testing.TB, rt runtime.Runtime) *hcsv2.Host {
+	h, err := _getHost(rt, getTransport())
+	if err != nil {
+		t.Helper()
+		t.Fatalf("could not get host: %v", err)
+	}
+
+	return h
+}
+
+func _getHost(rt runtime.Runtime, tp transport.Transport) (*hcsv2.Host, error) {
```

Comment:
I think we don't typically use underscores for function names in go

Reply:
I had that `_getHost`, and a wrapper `getHost`, and I didnt know how to differentiate them.
I've seen all underscore versions (`getHost` vs `gethost`), but I guess I can go with `getHostErr`...